runtime/vm/CRIUHelpers.cpp

```diff
@@ -602,6 +619,58 @@ runInternalJVMRestoreHooks(J9VMThread *currentThread, const char **nlsMsgFormat)
 		}
 	}
 
+#if JAVA_SPEC_VERSION >= 20
+	if (0 != vm->checkpointState.checkpointCPUCount) {
+		/* Only reset j.l.VirtualThread.ForkJoinPool.parallelism if jdk.virtualThreadScheduler.parallelism is not set. */
+		PORT_ACCESS_FROM_VMC(currentThread);
+		UDATA cpuCount = j9sysinfo_get_number_CPUs_by_type(J9PORT_CPU_TARGET);
+		if (cpuCount < 1) {
+			/* This matches ForkJoinPool default constructor with Runtime.getRuntime().availableProcessors(). */
+			cpuCount = 1;
+		}
+		if (cpuCount != vm->checkpointState.checkpointCPUCount) {
+#define JAVA_LANG_VIRTUALTHREAD "java/lang/VirtualThread"
+			J9Class *jlVirtualThreadClass = peekClassHashTable(currentThread, vm->systemClassLoader, (U_8 *)JAVA_LANG_VIRTUALTHREAD, LITERAL_STRLEN(JAVA_LANG_VIRTUALTHREAD));
+#undef JAVA_LANG_VIRTUALTHREAD
+			void *defaultSchedulerAddress = staticFieldAddress(currentThread, jlVirtualThreadClass, (U_8*)"DEFAULT_SCHEDULER", strlen("DEFAULT_SCHEDULER"), (U_8*)"Ljava/util/concurrent/ForkJoinPool;", strlen("Ljava/util/concurrent/ForkJoinPool;"), NULL, NULL, 0, NULL);
```

Comment:
can you put the code that gets the address and does the update of the parallelism field in a helper in VMHelpers, we will need it for other things

Reply:
Yeah, added `getStaticFieldObject`/`findinstanceFieldOffset`/`setObjectFieldI32` VMHelper methods.